modules/apps/portal-security/portal-security-iframe-sanitizer-test/src/testIntegration/java/com/liferay/portal/security/iframe/sanitizer/internal/test/IFrameSanitizerImplTest.java

```diff
@@ -35,139 +47,204 @@ public class IFrameSanitizerImplTest {
 	public static final AggregateTestRule aggregateTestRule =
 		new LiferayIntegrationTestRule();
 
+	@Before
+	public void setUp() throws Exception {
+		_companyId = TestPropsValues.getCompanyId();
+	}
+
+	@After
+	public void tearDown() throws Exception {
+		String filterString = StringBundler.concat(
+			"(&(service.factoryPid=", _CONFIGURATION_PID, ".scoped",
+			")(companyId=", _companyId, "))");
+
+		Configuration[] configurations = _configurationAdmin.listConfigurations(
+			filterString);
+
+		if (ArrayUtil.isNotEmpty(configurations)) {
+			Configuration configuration = configurations[0];
+
+			if (configuration != null) {
+				configuration.delete();
+			}
+		}
```

Comment:
Shouldn't you save/restore any pre-existing IFrameConfiguration? Otherwise, running this on a developer environment will mess up the configurations.

Why not change the _withConfiguration impl to do just that using your new logic?
```
 private void _withConfiguration(
			long companyId, boolean enabled, boolean removeIFrameTags,
			String sandboxAttributeValues,
			UnsafeRunnable<Exception> unsafeRunnable)
		throws Exception {

		Configuration oldConfiguration = ...;

		try {
				_updateCompanyConfiguration(...);

				unsafeRunnable.run();
		} finally {
				// ... restore oldConfiguration
		}  
}
```

Reply:
On other hand, the implementation of will complicate the test with several companies

Reply:
I don't understand ... can you elaborate?